core/domain/skill_domain.py

```diff
@@ -653,3 +653,36 @@ def to_dict(self):
             'skill_model_created_on': self.skill_model_created_on,
             'skill_model_last_updated': self.skill_model_last_updated
         }
+
+
+class UserSkillMastery(object):
+    """Domain object for a user's mastery of a particular skill."""
+
+    def __init__(self, model_id, user_id, skill_id, degree_of_mastery):
+        """Constructs a SkillMastery domain object.
+
+        Args:
+            model_id: str. The model id corresponding to the user and
+                the skill.
+            user_id: str. The user id of the logged in user.
+            skill_id: str. The id of the skill.
+            degree_of_mastery: float. The progress of user for the
+                corresponding skill.
+        """
+        self.id = model_id
+        self.user_id = user_id
+        self.skill_id = skill_id
+        self.degree_of_mastery = degree_of_mastery
+
+    def to_dict(self):
```

Comment:
This method should have a dedicated unit test in skill_domain_test.

Reply:
okay